clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -17,16 +17,18 @@ import java.net.URI
 import java.time.format.DateTimeFormatter
 import java.time.{LocalDateTime, ZoneOffset}
 import java.util.UUID
+import org.apache.commons.lang3.StringUtils
 
 trait RangeGetter extends Serializable {
 
   /** @return all rangeIDs in metarange of commitID on repo.
    */
   def getRangeIDs(commitID: String, repo: String): Iterator[String]
 
-  /** @return all object addresses in range rangeID on repo
+  /** @return all object addresses in range rangeID on repo.
+   *  Absolute addresses are ignored, unless they are under the given storage namespace.
```

Comment:
* I would add here that the ignored addresses are of imported objects, and an explanation of absolute within the namespace and how di they get there
* Semantically, this method is getRangeAddressesOfDeltableObjects consider reflecting that in naming